lib/core/test/unit/Cardano/Wallet/Primitive/Types/TokenMapSpec.hs

```diff
@@ -255,6 +261,19 @@ prop_subtract_invariant m1 m2 = property $
   where
     Just result = TokenMap.subtract m1 m2
 
+prop_difference_zero :: TokenMap -> Property
+prop_difference_zero x =
+    x `difference` TokenMap.empty === x
+
+prop_difference_leq :: TokenMap -> TokenMap -> Property
+prop_difference_leq x y = property $
+    x `difference` y `leq` x
```

Comment:
Should the reverse not also be true?

I.e.:

```haskell
x `difference` y `leq` y
```

And I'm guessing there could be a symmetric property:

```hs
x `difference` y = y `difference` X
```

Reply:
If `x = {a -> 1}` and `y = {}`, then `x - y = {a -> 1}`, which is larger than `{}`

```haskell
       Falsifiable (after 8 tests and 8 shrinks):
         TokenMap (fromList [(UnsafeTokenPolicyId (Hash "\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170\170"),NonEmptyMap {least = (UnsafeTokenName "TokenA",TokenQuantity 1), rest = fromList []})])
         TokenMap (fromList [])
```